plugins/scaffolder-backend-module-gitlab/src/actions/gitlabGroupEnsureExists.ts

```diff
@@ -90,17 +90,24 @@ export const createGitlabGroupEnsureExistsAction = (options: {
         );
         if (!subGroup) {
           ctx.logger.info(`creating missing group ${fullPath}`);
-          parentId = (
-            await api.Groups.create(
-              name,
-              slug,
-              parentId
-                ? {
-                    parentId: parentId,
-                  }
-                : {},
-            )
-          )?.id;
+
+          parentId = await ctx.checkpoint({
```

Comment:
This action really needs some love to refactor these reassignment too, but can wait until later.

Reply:
Agree, this one is a bit tricky, didn't want to break it.
I even didn't know that it's possible to do such circular reassignment lol